redisvl/query/query.py

```diff
@@ -1,234 +1,179 @@
-            limit (int): The max number of results to include including the offset
-
-        Raises:
-            TypeError: If first or limit are NOT integers.
-        """
-        if not isinstance(first, int) or not isinstance(limit, int):
-            raise TypeError("Paging params must both be integers")
-
-        self._first = first
-        self._limit = limit
+    @property
+    def filter(self) -> FilterExpression:
+        """The filter expression for the query."""
+        return self._filter_expression
 
     @property
-    def query(self) -> Query:
-        raise NotImplementedError
+    def query(self) -> "BaseQuery":
+        """Return self as the query object."""
+        return self
 
     @property
     def params(self) -> Dict[str, Any]:
-        return {}
+        """Return the query parameters."""
+        return self._params
 
 
-class CountQuery(BaseQuery):
+class FilterQuery(BaseQuery):
     def __init__(
         self,
-        filter_expression: FilterExpression,
+        filter_expression: Optional[FilterExpression] = None,
+        return_fields: Optional[List[str]] = None,
+        num_results: int = 10,
         dialect: int = 2,
+        sort_by: Optional[str] = None,
+        in_order: bool = False,
         params: Optional[Dict[str, Any]] = None,
     ):
-        """A query for a simple count operation provided some filter expression.
+        """A query for running a filtered search with a filter expression.
 
         Args:
-            filter_expression (FilterExpression): The filter expression to query for.
+            filter_expression (Optional[FilterExpression]): The optional filter
+                expression to query with. Defaults to '*'.
+            return_fields (Optional[List[str]], optional): The fields to return.
+            num_results (Optional[int], optional): The number of results to return. Defaults to 10.
+            dialect (int, optional): The query dialect. Defaults to 2.
+            sort_by (Optional[str], optional): The field to order the results by. Defaults to None.
+            in_order (bool, optional): Requires the terms in the field to have the same order as the terms in the query filter. Defaults to False.
             params (Optional[Dict[str, Any]], optional): The parameters for the query. Defaults to None.
 
         Raises:
             TypeError: If filter_expression is not of type redisvl.query.FilterExpression
+        """
+        if filter_expression:
+            self._filter_expression = filter_expression
+        if params:
+            self._params = params
 
-        .. code-block:: python
+        self._num_results = num_results
 
-            from redisvl.query import CountQuery
-            from redisvl.query.filter import Tag
+        # Initialize the base query with the full query string constructed from the filter expression
+        query_string = self._build_query_string()
+        super().__init__(query_string)
```

Comment:
we should maybe call out somewhere that we are extending the RedisQuery now which is why it is like this